src/python/gem5/components/boards/abstract_system_board.py

```diff
@@ -48,7 +56,7 @@ def __init__(
         clk_freq: str,
         processor: "AbstractProcessor",
         memory: "AbstractMemorySystem",
-        cache_hierarchy: "AbstractCacheHierarchy",
+        cache_hierarchy: Optional["AbstractCacheHierarchy"] = None,
```

Comment:
This optional should probably be documented why it's optional somewhere...

Reply:
It's optional because AbstractBoard's is optional, and some subclasses of AbstractSystemBoard want it optional (and treat it like it is)

Reply:
should I add that note to a comment above the argument?